server/runtime/src/main/java/org/infinispan/server/logging/events/ServerEventLogger.java

```diff
@@ -72,7 +69,23 @@ void textLog(EventLogLevel level, EventLogCategory category, String message) {
    }
 
    void eventLog(ServerEventImpl event) {
-      getEventCache().putAsync(Util.threadLocalRandomUUID(), event)
+      if (eventCache == null) {
+         if (!cacheManager.isRunning(EVENT_LOG_CACHE)) {
+            // Cache is not running yet, we can't block this thread so offload blocking thread to log the message
+            blockingManager.runBlocking(() -> {
+               eventCache = cacheManager.getCache(EVENT_LOG_CACHE);
```

Comment:
Do we need any visibility change for `eventCache`?

Reply:
Naw, at worst case if a thread can't see it, it just calls getCache again.

Reply:
And it is using putAsync, so ordering was never guaranteed.